src/main/java/org/influxdb/BatchOptions.java

```diff
@@ -0,0 +1,90 @@
+package org.influxdb;
+
+import org.influxdb.dto.Point;
+import org.influxdb.impl.BatchOptionsImpl;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.function.BiConsumer;
+
+/**
+ * BatchOptions are used to configure batching of individual data point writes
+ * into InfluxDB. See {@link InfluxDB#enableBatch(BatchOptions)}
+ */
+public interface BatchOptions {
+
+  BatchOptions DEFAULTS = BatchOptionsImpl.DEFAULTS;
```

Comment:
This is kinda weird to have a backreference to the impl in the interface

Reply:
Why a interface for the options at all, decoupling is not achieved, why the extra effort ?

Reply:
I wanted to have javadoc documentation in a separate file and this thing is best I could do. Perhaps I may merge the code from the interface and the class into one BatchOptions class. Defaults would be reachable be BatchOptions.DEFAULTS. You think that this way it would be better ? Or do you have some other idea ?

Reply:
Yes i would prefer this. I think we have no benefit have splitting Batchoptions into Interface and Implementation.